### PR TITLE
add data storage screen

### PR DESCRIPTION
**Pull request by EntyR**

### CODE REVIEW

File: app/src/main/java/com/harman/roomdbapp/app/ui/fragments/RandomNumbersList.kt [hunk 1]
```diff
@@ -60,6 +68,10 @@ class RandomNumbersList : Fragment() {
                     Widget(
                         getString(R.string.record_gravity_fluctuation),
                         R.drawable.grav_widget
+                    ),
+                    Widget(
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/ui/fragments/RandomNumbersList.kt:72]: please check the text size, for now it is different on both cards

File: app/src/main/java/com/harman/roomdbapp/app/ui/fragments/DataStorageFragment.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.harman.roomdbapp.app.adapters.DocumentsAdapter
+import com.harman.roomdbapp.app.adapters.callbaks.SwipeToDeleteCallback
+import com.harman.roomdbapp.app.databinding.FragmentDataStorageBinding
+import com.harman.roomdbapp.domain.model.Document
+
+class DataStorageFragment : Fragment() {
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/ui/fragments/DataStorageFragment.kt:15]: back button handling

File: app/src/main/res/layout/data_storage_item.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    tools:background="#EAEAEA">
+
+    <androidx.appcompat.widget.AppCompatButton
```
Comment (AlexeyKolosov) [inline app/src/main/res/layout/data_storage_item.xml]: please add marquee for text

File: app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/lnBottomSheet"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+
+    <TextView
```
Comment (AlexeyKolosov) [inline app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml:10]: please add marquee for long text

File: app/src/main/java/com/harman/roomdbapp/app/ui/fragments/BottomSheetFragment.kt [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.harman.roomdbapp.app.R
+import com.harman.roomdbapp.app.adapters.DocumentsEntryAdapter
+import com.harman.roomdbapp.app.databinding.FragmentBottomSheetListDialogBinding
+import com.harman.roomdbapp.domain.model.GravityRecord
+
+const val DOC_NAME = "doc_name"
+
+class BottomSheetFragment : BottomSheetDialogFragment() {
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/ui/fragments/BottomSheetFragment.kt:21]: please check: if we start swiping this bottom sheet down, we can stop at some state (maybe half expanded or something). We need to fix this. Expected: user can swipe this bottom sheet by one gesture.

File: app/src/main/java/com/harman/roomdbapp/app/adapters/callbaks/SwipeToDeleteCallback.kt [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.harman.roomdbapp.app.adapters.callbaks
+
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.RecyclerView
+
+class SwipeToDeleteCallback(private val deleteCallback: (position: Int) -> Unit) : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/adapters/callbaks/SwipeToDeleteCallback.kt]: another items should be moved to free space
Screenshot_1649318293

File: app/src/main/java/com/harman/roomdbapp/app/ui/fragments/RandomNumbersList.kt (hunk 1 above)
Comment (EntyR) [reply app/src/main/java/com/harman/roomdbapp/app/ui/fragments/RandomNumbersList.kt:72]: done

File: app/src/main/java/com/harman/roomdbapp/app/ui/fragments/DataStorageFragment.kt (hunk 2 above)
Comment (EntyR) [reply app/src/main/java/com/harman/roomdbapp/app/ui/fragments/DataStorageFragment.kt:15]: done

File: app/src/main/res/layout/data_storage_item.xml (hunk 3 above)
Comment (EntyR) [reply app/src/main/res/layout/data_storage_item.xml]: done

File: app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml (hunk 4 above)
Comment (EntyR) [reply app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml:10]: done

File: app/src/main/java/com/harman/roomdbapp/app/adapters/callbaks/SwipeToDeleteCallback.kt (hunk 6 above)
Comment (EntyR) [reply app/src/main/java/com/harman/roomdbapp/app/adapters/callbaks/SwipeToDeleteCallback.kt]: fixed

File: app/src/main/java/com/harman/roomdbapp/app/ui/fragments/BottomSheetFragment.kt (hunk 5 above)
Comment (EntyR) [reply app/src/main/java/com/harman/roomdbapp/app/ui/fragments/BottomSheetFragment.kt:21]: done

File: app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml (hunk 4 above)
Comment (AlexeyKolosov) [reply app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml:10]: Didn't you forget to add android:ellipsize="marquee" ? here

File: app/src/main/java/com/harman/roomdbapp/app/adapters/callbaks/SwipeToDeleteCallback.kt (hunk 6 above)
Comment (AlexeyKolosov) [reply app/src/main/java/com/harman/roomdbapp/app/adapters/callbaks/SwipeToDeleteCallback.kt]: Well done) works perfectly

**Comment by AlexeyKolosov** [comment]
Seems that this issue has not been fixed, see skype
image

**Comment by AlexeyKolosov** [comment]
Please check if quickly remove some items, there can be some free space

https://user-images.githubusercontent.com/48712070/162474858-8f38d21a-05c8-42b9-b15b-1db9f9e7ab2d.mp4

s

File: app/src/main/java/com/harman/roomdbapp/app/adapters/DocumentsAdapter.kt [hunk 7]
```diff
@@ -49,4 +58,10 @@ class DocumentsAdapter(
             itemPressedCallback(currentList[position].docName)
         }
     }
+
+    fun deleteIem(position: Int) {
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/adapters/DocumentsAdapter.kt]: typo

File: app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml (hunk 4 above)
Comment (EntyR) [reply app/src/main/res/layout/fragment_bottom_sheet_list_dialog.xml:10]: yep. now added

**Review by AlexeyKolosov — APPROVED** [review]

**Pull request merged**